### PR TITLE
feat: replace partner identification mechanism

### PR DESCRIPTION
**Pull request by OnyekaIjeh**

### CODE REVIEW

File: test/integration/PartnerRegistrar.spec.ts [hunk 1]
```diff
@@ -34,6 +33,7 @@ const tldAsSha3 = ethers.utils.id('rsk');
 const initialSetup = async () => {
   const signers = await ethers.getSigners();
   const owner = signers[0];
+  const partner = signers[0];
```
Comment (sergio-jara) [inline test/integration/PartnerRegistrar.spec.ts:36]: the same as the owner? 👀

File: test/integration/PartnerRenewer.spec.ts [hunk 2]
```diff
@@ -38,6 +36,7 @@ const initialSetup = async () => {
   const signers = await ethers.getSigners();
   const owner = signers[0];
   const partner = signers[0];
+  const partnerOwnerAccount = signers[0];
```
Comment (sergio-jara) [inline test/integration/PartnerRenewer.spec.ts:39]: same 👀

**Review by sergio-jara — CHANGES_REQUESTED** [review]

File: contracts/Registrar/PartnerRegistrar.sol [hunk 3]
```diff
@@ -123,36 +246,52 @@ contract PartnerRegistrar is IBaseRegistrar, Ownable {
...
+        address partner
     ) private returns (uint256) {
+        IPartnerConfiguration partnerConfiguration = _getPartnerConfiguration(
+            partner
+        );
+
         bytes32 label = keccak256(abi.encodePacked(name));
         require(
-            name.strlen() >= _getPartnerConfiguration().getMinLength(),
+            name.strlen() >= partnerConfiguration.getMinLength(),
```
[the remark this answers is not in the document]
Comment (OnyekaIjeh) [reply contracts/Registrar/PartnerRegistrar.sol:285]: Haven't merged to main

File: contracts/Registrar/PartnerRegistrar.sol [hunk 4]
```diff
@@ -123,36 +246,52 @@ contract PartnerRegistrar is IBaseRegistrar, Ownable {
...
             "Name too short"
         );
 
         require(
-            name.strlen() <= _getPartnerConfiguration().getMaxLength(),
+            name.strlen() <= partnerConfiguration.getMaxLength(),
```
[the remark this answers is not in the document]
Comment (OnyekaIjeh) [reply contracts/Registrar/PartnerRegistrar.sol:290]: See comment above

File: contracts/Registrar/PartnerRegistrar.sol [hunk 5]
```diff
@@ -54,6 +59,91 @@ contract PartnerRegistrar is IBaseRegistrar, Ownable {
...
+        | name       | variable size - offset 108
+    */
+
+    /// @notice ERC-677 token fallback function.
+    /// @dev Follow 'Register encoding' to execute a one-transaction regitration.
+    /// @param from token sender.
+    /// @param value amount of tokens sent.
+    /// @param data data associated with transaction.
+    /// @return true if successfull.
+    function tokenFallback(
```
[the remark this answers is not in the document]
Comment (OnyekaIjeh) [reply contracts/Registrar/PartnerRegistrar.sol:78]: I think there's a receiver interface for this. Will investigate.

File: contracts/Renewer/PartnerRenewer.sol [hunk 6]
```diff
@@ -74,25 +138,24 @@ contract PartnerRenewer is IBaseRenewer, Ownable {
...
-    function _getPartnerConfiguration()
-        private
-        view
-        returns (IPartnerConfiguration)
-    {
-        return _partnerManager.getPartnerConfiguration(msg.sender);
+    function _getPartnerConfiguration(
+        address partner
+    ) private view returns (IPartnerConfiguration) {
+        return _partnerManager.getPartnerConfiguration(partner);
```
[the remark this answers is not in the document]
Comment (OnyekaIjeh) [reply contracts/Renewer/PartnerRenewer.sol:159]: Exactly the same, but do we need an abstract contract for just this function?

File: test/integration/PartnerRegistrar.spec.ts (hunk 1 above)
Comment (OnyekaIjeh) [reply test/integration/PartnerRegistrar.spec.ts:36]: Will change 👀

File: test/integration/PartnerRenewer.spec.ts (hunk 2 above)
Comment (OnyekaIjeh) [reply test/integration/PartnerRenewer.spec.ts:39]: Will change 👀

File: contracts/Renewer/PartnerRenewer.sol (hunk 6 above)
[the remark this answers is not in the document]
Comment (sergio-jara) [reply contracts/Renewer/PartnerRenewer.sol:151]: nah

**Review by sergio-jara — APPROVED** [review]
great job

**Pull request merged**